src/redismodule.h

```diff
@@ -268,56 +265,35 @@ typedef enum {
- * distinctly deletion, overwrite or read-only would be marked as RW. */
-#define REDISMODULE_CMD_KEY_RO (1ULL<<0) /* Read-Only - Reads the value of the
-                                          * key, but doesn't necessarily returns
-                                          * it. */
-#define REDISMODULE_CMD_KEY_RW (1ULL<<1) /* Read-Write - Modifies the data
-                                          * stored in the value of the key or
-                                          * its metadata. */
-#define REDISMODULE_CMD_KEY_OW (1ULL<<2) /* Overwrite - Overwrites the data
-                                          * stored in the value of the key. */
-#define REDISMODULE_CMD_KEY_RM (1ULL<<3) /* Deletes the key. */
-/* The following refer to user data inside the value of the key, not the metadata
- * like LRU, type, cardinality. It refers to the logical operation on the user's
- * data (actual input strings / TTL), being used / returned / copied / changed,
- * It doesn't refer to modification or returning of metadata (like type, count,
- * presence of data). Any write that's not INSERT or DELETE, would be an UPDATE.
- * Each key-spec may have one of the writes with or without access, or none: */
-#define REDISMODULE_CMD_KEY_ACCESS (1ULL<<4) /* Returns, copies or uses the user
-                                              * data from the value of the
-                                              * key. */
-#define REDISMODULE_CMD_KEY_UPDATE (1ULL<<5) /* Updates data to the value, new
-                                              * value may depend on the old
-                                              * value. */
-#define REDISMODULE_CMD_KEY_INSERT (1ULL<<6) /* Adds data to the value with no
-                                              * chance of modification or
-                                              * deletion of existing data. */
-#define REDISMODULE_CMD_KEY_DELETE (1ULL<<7) /* Explicitly deletes some content
-                                              * from the value of the key. */
-/* Other flags: */
-#define REDISMODULE_CMD_KEY_CHANNEL (1ULL<<8)     /* PUBSUB shard channel */
-#define REDISMODULE_CMD_KEY_INCOMPLETE (1ULL<<9)  /* Means that the keyspec
-                                                   * might not point out to all
-                                                   * keys it should cover */
+/* Key-spec flags. For details, see the documentation of
+ * RedisModule_SetCommandInfo and the key-spec flags in server.h. */
+#define REDISMODULE_CMD_KEY_RO (1ULL<<0)
+#define REDISMODULE_CMD_KEY_RW (1ULL<<1)
+#define REDISMODULE_CMD_KEY_OW (1ULL<<2)
+#define REDISMODULE_CMD_KEY_RM (1ULL<<3)
+#define REDISMODULE_CMD_KEY_ACCESS (1ULL<<4)
+#define REDISMODULE_CMD_KEY_UPDATE (1ULL<<5)
+#define REDISMODULE_CMD_KEY_INSERT (1ULL<<6)
+#define REDISMODULE_CMD_KEY_DELETE (1ULL<<7)
+#define REDISMODULE_CMD_KEY_CHANNEL (1ULL<<8)
+#define REDISMODULE_CMD_KEY_INCOMPLETE (1ULL<<9)
```

Comment:
we probably need to expose VARIABLE_FLAGS too.
which means that the flags provided by the key-spec may be different than the ones provided by the getkeys-api (more specific to the exact args passed)

Reply:
wanna add a variant of `RM_KeyAtPos` that takes flags as part of this PR? or the next one?

Reply:
This is the problem with keeping PRs for a long time. This PR was ready before VARIABLE_FLAGS and keyspec notes existed. :-) So I think we can postpone to the next PR adding the KeyAtPosWithFlags and making the keypos-api smarter by infering positions from keyspecs when possible.

Reply:
I understand the general problem and principle, not sure it's applicable in this case. 
I don't mind postponing this, but for the sake of arguing... 
The reason this PR was postponed was because its priority was demoted, and the reason the variable flags PR was promoted is because it became urgent. We had to resolve certain things before the release and it was OK to leave other out. 

I wouldn't want to add content to this PR if it was controversial and deciding on it or implementing it could have delayed the rest of this PR. 
But this api is kinda trivia. And i think certain things are better be judged together (so we can make sure we understand the final picture before merging) 
(unlike we what we had with the commands and functions projects that  kept changing again and again with each PR)

Reply:
On a second thought, it should be judged together with another change (ACL, and tests), so let's leave it out.

Reply:
VARIABLE_FLAGS added.